packages/gestalt/src/Box.js

```diff
@@ -80,96 +54,331 @@ extra runtime typechecks in the transformers for performance.
 
 */
 
+type UnionRefs = HTMLDivElement | HTMLAnchorElement;
```

Comment:
Interestingly, these types _are_ resolved correctly. It appears the dividing line is imported types.